### ISSUE TITLE
Fails to glue chunks together when uploading to a folder that doesn't yet exist

### ISSUE THREAD

**Issue by Ellpeck** [shiftpi/nextcloud-chunk-file-upload#22]
Hi, thanks so much for this library, it's been great so far!!

However, I've noticed an issue when trying to upload a file to a directory that doesn't exist yet.  Let's imagine I'm trying to upload `Test.txt` to `Uploads/Test.txt` in my user's home directory, but the `Uploads` directory doesn't exist yet. I'd expect it to be created automatically in the process of moving the finished chunks together. Instead, this error occurs:

```
Test.txt (chunk 1, complete = false): Failed to glue the chunks together
```

Further inspecting the resulting `Event` yields these HTTP response infos:
```
error 409
<?xml version="1.0" encoding="utf-8"?>
<d:error xmlns:d="DAV:" xmlns:s="http://sabredav.org/ns">
  <s:exception>Sabre\DAV\Exception\Conflict</s:exception>
  <s:message>The destination node is not found</s:message>
</d:error>
```

Thanks so much for any help and/or fixes!

**Comment by Ellpeck** [comment]
Update: I solved this myself using the following code. It seems like each subdirectory has to be created individually, as otherwise a similar error occurs. I don't know if this is the best (or a particularly elegant) solution, but maybe it helps for the official implementation!

```js
let currDir = "";
for (let dir of dest.split("/").slice(0, -1)) {
    currDir += `${dir}/`;
    try {
        await axios.request({
            method: 'mkcol',
            url: `${basePath}/files/${user}/${currDir}`,
            auth: {
                username: user,
                password: token
            }
        });
        if (!quiet)
            console.log(`Created directory ${currDir}`);
    } catch (error) {
        // this is fine since the directory likely already exists
    }
}
```

**Comment by shiftpi** [comment]
Hey Ell,
Thanks for reporting! You're absolutely right, creating files in non-existing directories is currently not supported by this library.
But as you already pointed out, it's also not too hard to add this and it would make sense imo. I will check if I can release a new version with this feature soonish.

**Comment by shiftpi** [comment]
Fixed in release v1.2.0
